### DISCUSSION TITLE
1.24.6&1.24.13 get_text different

### DISCUSSION THREAD

**Discussion by tenghian** [pymupdf/PyMuPDF#4032] [General]
![2024-11-09_3-31-06](https://github.com/user-attachments/assets/5d9b85b3-4040-4de6-9ca3-8e0974faf9d6)
Extraction of blocks in version 1.24.6 is perfect. How can I make version 1.24.13 work like 1.24.6? Thank you!

```
text_blocks = page.get_text("dict", flags=pymupdf.TEXTFLAGS_BLOCKS)["blocks"]
```
[pdf1246-cut.pdf](https://github.com/user-attachments/files/17683337/pdf1246-cut.pdf)
[pdf12413-cut.pdf](https://github.com/user-attachments/files/17683338/pdf12413-cut.pdf)

**Comment by JorjMcKie** [comment]
You cannot - except with your own code of course.

PyMuPDF is not deciding about block segmentation, this is a result of MuPDF's algorithms. The next MuPDF version 1.25.0 will bring significant improvements here. With a new text extraction option, MuPDF can be asked to search for recognizable page layout segments which will each be turned into a block for PyMuPDF.
That should bring back some of the earlier results.
In yet a subsequent version, MuPDF will also add guessing correct paragraph breaks and other structures.
But this is more in the future.
